`Plate.rename_well` is broken on Windows

Likely due to extra slashes in the URI to path conversion.

https://github.com/czbiohub-sf/iohub/blob/985b4ea45e9d2f82d8f5fc7d58a773d1c50a6e84/iohub/ngff/nodes.py#L1939-L1942

```py
         store_path = Path(
            str(self.zgroup.store_path).replace("file:", "")
        )  # zarr-python prepends file: for some reason
>       assert store_path.is_dir()
               ^^^^^^^^^^^^^^^^^^^
E       AssertionError
```

--- a/iohub/ngff/nodes.py
+++ b/iohub/ngff/nodes.py
@@ -1932,9 +1932,9 @@ class Plate(NGFFNode):
         # raises ValueError if old well does not exist
         # or if new well already exists
         if old not in self.zgroup:
-            raise ValueError(f"Well '{old}' does not exist.")
+            raise FileNotFoundError(f"Well '{old}' does not exist.")
         if new in self.zgroup:
-            raise ValueError(f"Well '{new}' already exists.")
+            raise FileExistsError(f"Well '{new}' already exists.")
 
         store_path = self.zgroup.store.root
         assert store_path.is_dir()

--- a/iohub/ngff/nodes.py
+++ b/iohub/ngff/nodes.py
@@ -1936,9 +1936,7 @@ class Plate(NGFFNode):
         if new in self.zgroup:
             raise ValueError(f"Well '{new}' already exists.")
 
-        store_path = Path(
-            str(self.zgroup.store_path).replace("file:", "")
-        )  # zarr-python prepends file: for some reason
+        store_path = self.zgroup.store.root
         assert store_path.is_dir()
 
         old_path = store_path / old

--- a/tests/ngff/test_ngff.py
+++ b/tests/ngff/test_ngff.py
@@ -580,11 +580,11 @@ def test_rename_well(channels_and_random_5d, arr_name, version):
         assert "4" in [col.name for col in dataset.metadata.columns]
 
         # destination exists
-        with pytest.raises(ValueError):
+        with pytest.raises(FileExistsError):
             dataset.rename_well("B/2", "C/4")
 
         # source doesn't exist
-        with pytest.raises(ValueError):
+        with pytest.raises(FileNotFoundError):
             dataset.rename_well("Q/1", "Q/2")
 
         # invalid well names

